# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/IonTypes.h
# Commit: cf96078694ae
# Full Hash: cf96078694aea74cae4557f683a76c6de6ba9cf1
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Rename MIRType::RefOrNull to 'WasmAnyRef'. r=yury
#   
#   This is a mechanical change. The previous name 'RefOrNull' was too
#   generic, and it wasn't clear that the type was actually a wasm::AnyRef
#   type.
# ==============================================================================

diff -r 662cb63c42d7 -r cf96078694ae js/src/jit/IonTypes.h
--- a/js/src/jit/IonTypes.h	Fri Aug 04 14:06:49 2023 +0000
+++ b/js/src/jit/IonTypes.h	Fri Aug 04 14:06:49 2023 +0000
@@ -515,7 +515,7 @@
   Slots,         // A slots vector
   Elements,      // An elements vector
   Pointer,       // An opaque pointer that receives no special treatment
-  RefOrNull,     // Wasm Ref/AnyRef/NullRef: a raw JSObject* or a raw (void*)0
+  WasmAnyRef,     // Wasm Ref/AnyRef/NullRef: a raw JSObject* or a raw (void*)0
   StackResults,  // Wasm multi-value stack result area, which may contain refs
   Shape,         // A Shape pointer.
   Last = Shape
@@ -605,7 +605,7 @@
     case MIRType::Simd128:
       return 16;
     case MIRType::Pointer:
-    case MIRType::RefOrNull:
+    case MIRType::WasmAnyRef:
       return sizeof(uintptr_t);
     default:
       MOZ_CRASH("MIRTypeToSize - unhandled case");
@@ -656,8 +656,8 @@
       return "Elements";
     case MIRType::Pointer:
       return "Pointer";
-    case MIRType::RefOrNull:
-      return "RefOrNull";
+    case MIRType::WasmAnyRef:
+      return "WasmAnyRef";
     case MIRType::StackResults:
       return "StackResults";
     case MIRType::Shape: